nasa-gibs/worldview
# Init compare B side boolean if permalinked, revert c058658

## Description

Fixes #3721 .

- [x] Set `bStatesInitiated` to true if permalink is compare mode. This allows relative -7 days compare B side date as expected if compare mode started for the first time.
- [x] Revert "remove init second date logic and update test". This reverts commit c058658.

## How To Test

1. http://localhost:3000/?t=2020-04-20
2. Start comparison mode
3. See B date is 2020-04-13
4. Select B side and exit Compare mode
5. See selected date is 2020-04-13

Related to #3708

Initially A Tab Selected
1. http://localhost:3000/?v=-115.93752715236366,35.49962860732,-113.39248613039783,36.78279485598542&l=Reference_Labels_15m,Coastlines_15m,MODIS_Terra_CorrectedReflectance_Bands721,VIIRS_SNPP_CorrectedReflectance_TrueColor,MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&l1=Reference_Labels_15m,Coastlines_15m,MODIS_Terra_CorrectedReflectance_Bands721,MODIS_Terra_CorrectedReflectance_TrueColor&lg1=true&ca=true&cv=48&t=2001-09-06-T15%3A20%3A13Z&t1=2021-09-04-T15%3A20%3A13Z
2. Switch to B Tab
3. Exit Comparison Mode
4. Verify expected result of date/imagery being 2021-SEP-04


Initially B Tab Selected

1. http://localhost:3000/?v=-115.93752715236366,35.49962860732,-113.39248613039783,36.78279485598542&l=Reference_Labels_15m,Coastlines_15m,MODIS_Terra_CorrectedReflectance_Bands721,VIIRS_SNPP_CorrectedReflectance_TrueColor,MODIS_Terra_CorrectedReflectance_TrueColor&lg=true&l1=Reference_Labels_15m,Coastlines_15m,MODIS_Terra_CorrectedReflectance_Bands721,MODIS_Terra_CorrectedReflectance_TrueColor&lg1=true&ca=false&cv=48&t=2001-09-06-T15%3A20%3A13Z&t1=2021-09-04-T15%3A20%3A13Z
2. Exit Comparison Mode
3. Verify expected result of date/imagery being 2021-SEP-04



## PR Submission Checklist

This is simply a reminder of what we are going to look for before merging your code.

- I have read the [CONTRIBUTING](https://github.com/nasa-gibs/worldview/blob/main/.github/CONTRIBUTING.md) doc
- I have added necessary documentation (if applicable)
- I have added tests that prove my fix is effective or that my feature works (if applicable)
- Any dependent changes have been merged and published in downstream modules (if applicable)

@nasa-gibs/worldview
